near/src/config.rs

```diff
@@ -544,6 +548,14 @@ impl From<&str> for GenesisConfig {
                 config.protocol_version, PROTOCOL_VERSION
             ));
         }
+        let num_shards = config.block_producers_per_shard.len();
+        let num_chunk_producers: ValidatorId = config.block_producers_per_shard.iter().sum();
+        if num_chunk_producers != max(config.num_block_producers, num_shards) {
+            panic!(format!(
```

Comment:
Does near unit --num-shards work after this change?

Reply:
and `near testnet -s 8`

e.g. prob need to fix some of the testing configs initialization